IdentityCore/src/cache/serializers/MSIDKeyedArchiverSerializer.m

```diff
@@ -64,8 +85,13 @@ - (MSIDLegacyTokenCacheItem *)deserialize:(NSData *)data className:(Class)classN
     
     NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:data];
     // Maintain backward compatibility with ADAL.
-    [unarchiver setClass:MSIDUserInformation.class forClassName:@"ADUserInformation"];
     [unarchiver setClass:className forClassName:@"ADTokenCacheStoreItem"];
+    for (NSString *c in _defaultDecodeClassMap)
+    {
+        [unarchiver setClass:_defaultDecodeClassMap[className] forClassName:c];
+    }
+    [unarchiver setClass:MSIDUserInformation.class forClassName:@"ADUserInformation"];
```

Comment:
this one is already inside _defaultDecodeClassMap, why do we repeat it?

Reply:
:+1: Good catch

Reply:
fixed